### ISSUE TITLE
SSA validation FAIL tensor concat_19_0'.1__mye152's uses list size does not match the actual use in the program:1 vs. 0

### ISSUE THREAD

**Issue by handoku** [NVIDIA/TensorRT#2009]
## Description
when use trtexec to convert a model (eventhough model's input shape is fixed) to trt engine, I get this myelin error：
```bash
05/26/2022-17:09:25] [W] [TRT] parsers/onnx/ShapedWeights.cpp:171: Weights MPS_VAR_1/l2_normalize_2__333 has been transposed with permutation of (1, 0)! If you plan on overwriting the weights with the Refitter API, the new weights must be pre-transposed.
[05/26/2022-17:09:25] [I] Finish parsing network model
[05/26/2022-17:09:26] [I] [TRT] [MemUsageChange] Init cuBLAS/cuBLASLt: CPU +519, GPU +224, now: CPU 1029, GPU 512 (MiB)
[05/26/2022-17:09:26] [I] [TRT] [MemUsageChange] Init cuDNN: CPU +114, GPU +52, now: CPU 1143, GPU 564 (MiB)
[05/26/2022-17:09:26] [I] [TRT] Local timing cache in use. Profiling results in this builder pass will not be stored.
SSA validation FAIL: tensor concat_19_0'.1__mye152's uses list size does not match the actual use in the program:1 vs. 0

In tensor uses:
	op: Segmentation fault (core dumped)
```

To be honest, recently, I have encontered lots of `myelin errors` , especially something like `dims/size not match`. I understand some optimization pass need to fuse op, combine tensors, change tensors' layout/shape etc. And I understand trt 8.4 GA may will fix these problems. But I want a way to solve the issue ASAP.

I feel that this problem might caused by some sensitive op's related shape/attr.  Therefore，I want to know is there any way (even 
 a trick，like modifying onnx model) to avoid sensitive points，and to avoid this problem. And finally, In that way I can just export trt engine normally.

@nvpohanh Could you please help look into this, thanks!

## Environment
I am using docker image ngc-tensorrt:2202

**TensorRT Version**:  8.2.3.4
**NVIDIA GPU**:  T4
**NVIDIA Driver Version**: 450.80.02 
**CUDA Version**: 11.6

## Relevant Files

[dynamic shape model](https://drive.google.com/file/d/1LU722NCwPf-mU2rV8HpEDnFOwyC2QVuc/view?usp=sharing)
[fixed shape model](https://drive.google.com/file/d/1saVGFSOXNwsBRm9J7wYlzoDpPeAXX23W/view?usp=sharing)

## Steps To Reproduce
dynamic shape:
``` bash
trtexec --onnx=dynamic.onnx --explicitBatch  --optShapes=\'SLOT_INPUT/ALL_IN_ONE:0\':170x1x6942  --buildOnly --saveEngine=dynamic.plan --workspace=12288  --device=2
```
fixed shape:
```bash
trtexec --onnx=fixed.onnx --explicitBatch --buildOnly --saveEngine=fixed.plan --workspace=12288 --device=2
```

**Comment by nvpohanh** [comment]
I can repro the failure using TRT 8.2. The issue is gone when I use TRT 8.4 GA internal release candidate, which will probably be released in June. Please stay tuned for TRT 8.4 GA release. Thanks for your patience.

**Comment by handoku** [comment]
> I can repro the failure using TRT 8.2. The issue is gone when I use TRT 8.4 GA internal release candidate, which will probably be released in June. Please stay tuned for TRT 8.4 GA release. Thanks for your patience.

Are you able to access myelin module code? if you can, could you please provide some detailed error info about this error?

The model I uploaded is simplified by my custom script with `onnx graphsurgeon`.  And before this simplification, it can be exported to trt normally. If you can provide some details, I could/might avoid this problem with some more modifications.

**Comment by nvpohanh** [comment]
TRT 8.4.1 (8.4 GA) has been released and will be part of 22.07 container. I am going to close this for now. Thanks

I don't have much insight to the myelin module, either. Will need to ask internally if you would like detailed answer.

**Comment by handoku** [comment]
> 

Thanks. The problem was caused by optimization for `concat/slice`, I fixed it by add some memory access op(likes `gather`) to prevent myelin's optimization.

And also I have tried with trt 8.4.1, the issue is gone.

**Comment by handoku** [comment]
@nvpohanh Would you please reopen this issue. Unfortunately, we got this error when convert [model.onnx](https://drive.google.com/file/d/1vkZK_znWFZbCVuof8yF8AcE4JRMrq7P4/view?usp=sharing) to engine using trt 8.4 GA.

Like I said before, we have got a lots of similar errors. the problem did not just disappear. When it show up, we have to take efforts to run tests. and sometimes we may avoid it with some workround, but it's really a little dirty and tricky. 

We hope TensorRT fix it from the start, sincerely

**Comment by handoku** [comment]
> @nvpohanh Would you please reopen this issue. Unfortunately, we got this error when convert [model.onnx](https://drive.google.com/file/d/1vkZK_znWFZbCVuof8yF8AcE4JRMrq7P4/view?usp=sharing) to engine using trt 8.4 GA.
> 
> Like I said before, we have got a lots of similar errors. the problem did not just disappear. When it show up, we have to take efforts to run tests. and sometimes we may avoid it with some workround, but it's really a little dirty and tricky.
> 
> We hope TensorRT fix it from the start, sincerely

@zerollzeng hi, could you please help to create a internal issue to track the failure.

**Comment by nvpohanh** [comment]
Reopen.

**Comment by nvpohanh** [comment]
I can repro the SSA failure using TRT 8.4.1.5. Filed internal tracked (id: 3693684). Will let you know if there is any update.

**Comment by nvpohanh** [comment]
This is fixed in TRT 8.5.

**Comment by ttyio** [comment]
Closing since no activity for more than 3 weeks, please reopen if you still have question, thanks!
